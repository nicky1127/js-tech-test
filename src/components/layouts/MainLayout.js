import React, { useState } from "react";
import PropTypes from "prop-types";
import clsx from "clsx";
import { connect } from "react-redux";
import { makeStyles } from "@material-ui/core/styles";
import {
  Box,
  Typography,
  FormControl,
  MenuItem,
  InputLabel,
  Select
} from "@material-ui/core";
import Loading from "./Loading";
import BetSlipPanel from "components/layouts/BetSlipPanel";
import { setOddsForamt } from "redux/_actions";
import labels from "constants/labels";

const constants = labels.MainLayout;

const useStyles = makeStyles(theme => ({
  root: {
    width: "100%",
    display: "flex",
    position: "relative"
  },
  contentWrapper: {
    width: "78%"
  },
  appBar: {
    height: "10vh",
    padding: "0 1.5rem",
    backgroundColor: "rgb(21,63,142)",
    display: "flex",
    alignItems: "center"
  },
  logo: {
    flex: "20 20 auto"
  },
  text: {
    color: "#fff",
    fontSize: "12px",
    padding: "0 10px",
    "&:hover": {
      cursor: "pointer"
    }
  },
  formControl: {
    width: "20%"
  },
  select: {
    borderRadius: 4,
    position: "relative",
    backgroundColor: theme.palette.background.paper,
    fontSize: "14px",
    textAlign: "center",
    "&:focus": {
      "& input": {
        color: "#b2ebf2"
      }
    }
  },
  label: {
    color: "#b2ebf2"
  }
}));

export const MainLayout = ({
  children,
  loading,
  isOddsDecimal,
  setOddsForamt
}) => {
  const classes = useStyles();

  const handleChange = evt => {
    setOddsForamt(evt.target.value);
  };

  return (
    <div className={clsx("mainlayout", classes.root)} data-testid="mainLayout">
      {loading && <Loading />}
      <Box className={clsx("contentWrapper", classes.contentWrapper)}>
        <Box className={clsx("appBar", classes.appBar)}>
          <Typography className={classes.text} component="span">
            {constants.links.MY_BETS}
          </Typography>
          <div className={classes.logo}></div>
          {/* <Typography className={classes.text} component="span">
            {constants.links.JOIN}
          </Typography>
          <Typography className={classes.text} component="span">
            {constants.links.LOG_IN}
          </Typography> */}
          <FormControl className={classes.formControl}>
            <InputLabel id="demo-simple-select-label" className={classes.label}>
              Odds Format
            </InputLabel>
            <Select
              className={classes.select}
              labelId="demo-simple-select-label"
              id="demo-simple-select"
              value={isOddsDecimal}
              onChange={handleChange}
            >
              <MenuItem value={true}>Decimal</MenuItem>
              <MenuItem value={false}>Fractional</MenuItem>
            </Select>
          </FormControl>
        </Box>
        {children}
      </Box>
      <BetSlipPanel />
    </div>
  );
};

MainLayout.defaultProps = {};

MainLayout.propTypes = {};

const mapStateToProps = state => {
  const { loading, isOddsDecimal } = state;
  return { loading, isOddsDecimal };
};

const ConnectedMainLayout = connect(mapStateToProps, { setOddsForamt })(
  MainLayout
);

export default ConnectedMainLayout;
