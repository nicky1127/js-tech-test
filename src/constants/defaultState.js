const state = {
  liveEvents: [],
  event: {},
  markets: {},
  isOddsDecimal: false,
  loading: false,
  error: {},
  selections: []
};

export default state;
