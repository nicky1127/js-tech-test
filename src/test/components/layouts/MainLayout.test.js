import React from "react";
import { render, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import MainLayout from "components/layouts/MainLayout";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import defaultState from "constants/defaultState";

beforeEach(() => {
  jest.clearAllMocks();
  cleanup();
});

const props = {};

const state = {
  ...defaultState
};
const mockStore = configureMockStore();
const store = mockStore(state);

describe("MainLayout", () => {
  it("Snapshot matchs", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MainLayout {...props} />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("MainLayout should render one [data-testid=mainLayout]", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <MainLayout {...props} />
      </Provider>
    );
    const container = getByTestId("mainLayout");
    expect(container).toBeInTheDocument();
  });
});
