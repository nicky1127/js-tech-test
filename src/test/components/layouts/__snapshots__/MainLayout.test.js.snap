// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainLayout Snapshot matchs 1`] = `
<div
  className="mainlayout makeStyles-root-1"
  data-testid="mainLayout"
>
  <div
    className="MuiBox-root MuiBox-root-9 contentWrapper makeStyles-contentWrapper-2"
  >
    <div
      className="MuiBox-root MuiBox-root-10 appBar makeStyles-appBar-3"
    >
      <span
        className="MuiTypography-root makeStyles-text-5 MuiTypography-body1"
      >
        My Bets
      </span>
      <div
        className="makeStyles-logo-4"
      />
      <div
        className="MuiFormControl-root makeStyles-formControl-6"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root makeStyles-label-8 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
          data-shrink={true}
          id="demo-simple-select-label"
        >
          Odds Format
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-select-7 MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <div
            aria-haspopup="listbox"
            aria-labelledby="demo-simple-select-label demo-simple-select"
            className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
            id="demo-simple-select"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            role="button"
            tabIndex={0}
          >
            Fractional
          </div>
          <input
            aria-hidden={true}
            className="MuiSelect-nativeInput"
            onAnimationStart={[Function]}
            onChange={[Function]}
            required={false}
            tabIndex={-1}
            value={false}
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSelect-icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
  <div
    className="betSlipPanel makeStyles-root-11 makeStyles-root-16"
    data-testid="betSlipPanel"
  >
    <div
      className="MuiBox-root MuiBox-root-18 makeStyles-header-12"
    >
      <div
        className="MuiBox-root MuiBox-root-19 makeStyles-numFlag-14"
      >
        0
      </div>
      <p
        className="MuiTypography-root makeStyles-headerText-15 MuiTypography-body1"
      >
        My Bet Slip
      </p>
    </div>
    <div
      className="MuiBox-root MuiBox-root-20 makeStyles-panelBody-13 makeStyles-panelBody-17"
    >
      <div
        className="makeStyles-root-21"
        data-testid="noSelection"
      >
        <p
          className="MuiTypography-root makeStyles-noSelectText-22 MuiTypography-body1"
        >
          You have no selections in your Bet Slip
        </p>
        <div
          className="MuiBox-root MuiBox-root-24 makeStyles-optionBox-23"
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            Go to In-Play bets
          </p>
        </div>
        <div
          className="MuiBox-root MuiBox-root-25 makeStyles-optionBox-23"
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            Build an Accumulator
          </p>
        </div>
        <div
          className="MuiBox-root MuiBox-root-26 makeStyles-optionBox-23"
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            Next Races
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
